### PR TITLE
Close Stale PR and Issues

### PR DESCRIPTION
**Pull request by gladjohn**
Close Stale PR and Issues

### CODE REVIEW

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+          close-issue-message: 'This issue was closed because it has been stalled for 5 days with no activity.'
+          close-pr-message: 'This PR was closed because it has been stalled for 10 days with no activity.'
+          stale-issue-label: 'no-issue-activity'
+          stale-pr-label: 'no-pr-activity'
+          exempt-pr-labels: 'awaiting-approval,work-in-progress'
+          only-labels: 'awaiting-feedback,awaiting-answers'
+          days-before-issue-stale: 120
+          days-before-pr-stale: 60
+          days-before-issue-close: 5
+          days-before-pr-close: 10
```
Comment (pmaytak) [inline .github/workflows/stale.yml:23]: Is this after the issue is marked stale or after it's created? Should it be 125 then?

File: .github/workflows/stale.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+          stale-issue-message: 'This issue is stale because it has been open 120 days with no activity. Remove stale label or comment or this will be closed in 5 days.'
+          stale-pr-message: 'This PR is stale because it has been open 60 days with no activity. Remove stale label or comment or this will be closed in 10 days.'
+          close-issue-message: 'This issue was closed because it has been stalled for 5 days with no activity.'
+          close-pr-message: 'This PR was closed because it has been stalled for 10 days with no activity.'
+          stale-issue-label: 'no-issue-activity'
+          stale-pr-label: 'no-pr-activity'
+          exempt-pr-labels: 'awaiting-approval,work-in-progress'
+          only-labels: 'awaiting-feedback,awaiting-answers'
+          days-before-issue-stale: 120
+          days-before-pr-stale: 60
```
Comment (pmaytak) [inline .github/workflows/stale.yml:21]: Hmm, I think it's too long. Should probably be like a week at most?

File: .github/workflows/stale.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  schedule:
+    - cron: '30 1 * * *'
+
+jobs:
+  stale:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/stale@v8
+        with:
+          stale-issue-message: 'This issue is stale because it has been open 120 days with no activity. Remove stale label or comment or this will be closed in 5 days.'
```
Comment (pmaytak) [inline .github/workflows/stale.yml:12]: How does this work - if the issue is marked as stale, and I remove the `stale` label, will it be marked stale again in 120 days?

File: .github/workflows/stale.yml (hunk 1 above)
Comment (gladjohn) [reply .github/workflows/stale.yml:23]: you have 5 days for issues and 10 days for prs that have been marked as stale, if you remove the stale label then it won't be closed

File: .github/workflows/stale.yml (hunk 2 above)
Comment (gladjohn) [reply .github/workflows/stale.yml:21]: we can agree upon a number

File: .github/workflows/stale.yml (hunk 3 above)
Comment (gladjohn) [reply .github/workflows/stale.yml:12]: Yes

Comment (pmaytak) [reply .github/workflows/stale.yml:12]: I'm just thinking we have issues that we should do but in the major version or something like that. So we'll be re-triaging every 3 months.

File: .github/workflows/stale.yml (hunk 2 above)
Comment (pmaytak) [reply .github/workflows/stale.yml:21]: We can discuss how this work in the stand up.

**Comment by trwalke** [comment]
Does this apply to draft PRs?

**Comment by gladjohn** [comment]
> Does this apply to draft PRs?

no

File: .github/workflows/stale.yml (hunk 3 above)
Comment (pmaytak) [inline .github/workflows/stale.yml:12]: As discussed, should just try this for PRs, not issues.

File: .github/workflows/stale.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+          stale-pr-message: 'This PR is stale because it has been open 60 days with no activity. Remove stale label or comment or this will be closed in 10 days.'
+          close-issue-message: 'This issue was closed because it has been stalled for 5 days with no activity.'
+          close-pr-message: 'This PR was closed because it has been stalled for 10 days with no activity.'
+          stale-issue-label: 'no-issue-activity'
+          stale-pr-label: 'no-pr-activity'
+          exempt-pr-labels: 'awaiting-approval,work-in-progress'
+          only-labels: 'awaiting-feedback,awaiting-answers'
+          days-before-issue-stale: 120
+          days-before-pr-stale: 20
```
Comment (pmaytak) [inline .github/workflows/stale.yml:21]: 7?

File: .github/workflows/stale.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    steps:
+      - uses: actions/stale@v8
+        with:
+          stale-issue-message: 'This issue is stale because it has been open 120 days with no activity. Remove stale label or comment or this will be closed in 5 days.'
+          stale-pr-message: 'This PR is stale because it has been open 60 days with no activity. Remove stale label or comment or this will be closed in 10 days.'
+          close-issue-message: 'This issue was closed because it has been stalled for 5 days with no activity.'
+          close-pr-message: 'This PR was closed because it has been stalled for 10 days with no activity.'
+          stale-issue-label: 'no-issue-activity'
+          stale-pr-label: 'no-pr-activity'
+          exempt-pr-labels: 'awaiting-approval,work-in-progress'
```
Comment (pmaytak) [inline .github/workflows/stale.yml:18]: I guess this is not really a perfect process - PRs with these labels can go stale too...

**Review by pmaytak — APPROVED** [review]

**Comment by gladjohn** [comment]
planning on closing this without merging. Need to come up with a different solution / reporting

**Pull request closed without merging**